test/e2e/tests/httproute_dualstack.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+//go:build e2e
+// +build e2e
+
+package tests
+
+import (
+	"os"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteDualStackTest)
+}
+
+var HTTPRouteDualStackTest = suite.ConformanceTest{
+	ShortName:   "HTTPRouteDualStack",
+	Description: "Test HTTPRoute support for IPv6 only, dual-stack, and IPv4 only services",
+	Manifests:   []string{"testdata/httproute-dualstack.yaml"},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		ipFamily := os.Getenv("IP_FAMILY")
+		ns := "gateway-conformance-infra"
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+
+		switch ipFamily {
+		case "ipv6":
+			t.Run("HTTPRoute to IPv6 only service", func(t *testing.T) {
+				runHTTPRouteTest(t, suite, ns, gwNN, "infra-backend-v1-httproute-ipv6", "/ipv6-only")
+			})
+		case "dual":
+			t.Run("HTTPRoute to IPv6 only service", func(t *testing.T) {
+				runHTTPRouteTest(t, suite, ns, gwNN, "infra-backend-v1-httproute-ipv6", "/ipv6-only")
+			})
+			t.Run("HTTPRoute to Dual-stack service", func(t *testing.T) {
+				runHTTPRouteTest(t, suite, ns, gwNN, "infra-backend-v1-httproute-dualstack", "/dual-stack")
+			})
+			t.Run("HTTPRoute to IPv4 only service", func(t *testing.T) {
+				runHTTPRouteTest(t, suite, ns, gwNN, "infra-backend-v1-httproute-ipv4", "/ipv4-only")
+			})
+		case "ipv4":
+			t.Run("HTTPRoute to IPv4 only service", func(t *testing.T) {
+				runHTTPRouteTest(t, suite, ns, gwNN, "infra-backend-v1-httproute-ipv4", "/ipv4-only")
+			})
+		default:
+			t.Skip("Skipping HTTPRouteDualStack test as IP_FAMILY is not set")
+		}
+	},
+}
+
+func runHTTPRouteTest(t *testing.T, suite *suite.ConformanceTestSuite, ns string, gwNN types.NamespacedName, routeName, path string) {
+	routeNN := types.NamespacedName{Name: routeName, Namespace: ns}
+	gwAddr := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+
+	expectedResponse := http.ExpectedResponse{
+		Request: http.Request{
+			Path: path,
+		},
+		Response: http.Response{
+			StatusCode: 200,
+		},
+		Namespace: ns,
+	}
+
+	http.MakeRequestAndExpectEventuallyConsistentResponse(t, suite.RoundTripper, suite.TimeoutConfig, gwAddr, expectedResponse)
```

Comment:
https://github.com/envoyproxy/gateway/blob/4b8c2f56dc45534c42b6513b126082f49a009658/test/e2e/tests/utils.go#L535

can be used to verify pod ip/port

Reply:
I'm still working on this test part. I'll post it separately as another PR.